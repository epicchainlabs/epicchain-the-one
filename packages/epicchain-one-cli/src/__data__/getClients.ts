import { getClients as getClientsBase } from '@neo-one/cli-common';
import { common, crypto } from '@neo-one/client-common';
import { NEOONEDataProvider } from '@neo-one/client-core';
import { constants } from '@neo-one/utils';

export const getClients = async (project: string) => {
  crypto.addPublicKey(
    common.stringToPrivateKey(constants.PRIVATE_NET_PRIVATE_KEY),
    common.stringToECPoint(constants.PRIVATE_NET_PUBLIC_KEY),
  );

  const config = await one.getProjectConfig(project);

  return getClientsBase(
    new NEOONEDataProvider({
      network: constants.LOCAL_NETWORK_NAME,
      rpcURL: `http://localhost:${config.network.port}/rpc`,
    }),
    constants.PRIVATE_NET_PRIVATE_KEY,
  );
};
