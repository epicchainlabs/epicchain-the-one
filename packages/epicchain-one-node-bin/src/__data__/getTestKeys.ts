import { common, crypto } from '@neo-one/client-common';
import { constants } from '@neo-one/utils';

export const getTestKeys = () => {
  const privateKey = common.stringToPrivateKey(constants.PRIVATE_NET_PRIVATE_KEY);
  const publicKey = common.stringToECPoint(constants.PRIVATE_NET_PUBLIC_KEY);
  crypto.addPublicKey(privateKey, publicKey);

  const standbyValidator = common.ecPointToString(publicKey);
  const address = common.uInt160ToString(crypto.privateKeyToScriptHash(privateKey));
  const privateKeyString = common.privateKeyToString(privateKey);

  return { standbyValidator, address, privateKeyString };
};
