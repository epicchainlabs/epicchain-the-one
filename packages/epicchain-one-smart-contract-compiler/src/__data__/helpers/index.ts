export * from './compile';
export * from './createContext';
export * from './execute';
export * from './extractors';
export * from './startNode';
export * from './transpileAndExecute';
export * from './utils';
