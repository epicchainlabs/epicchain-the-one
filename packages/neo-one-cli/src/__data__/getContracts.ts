import { Contract } from '@neo-one/client-common';
import { Client } from '@neo-one/client-full-core';

export const getContracts = async (client: Client, networkName: string): Promise<readonly Contract[]> => {
  const readClient = client.read(networkName);
  const currentIndex = await readClient.getBlockCount();

  const mutableContracts: Contract[] = [];
  // tslint:disable-next-line no-loop-statement
  for await (const block of readClient.iterBlocks({ indexStart: 0, indexStop: currentIndex })) {
    // tslint:disable-next-line no-loop-statement
    for (const transaction of block.transactions) {
      if (transaction.type === 'InvocationTransaction' && transaction.invocationData.contracts.length > 0) {
        mutableContracts.push(...transaction.invocationData.contracts);
      }
    }
  }

  return mutableContracts;
};
