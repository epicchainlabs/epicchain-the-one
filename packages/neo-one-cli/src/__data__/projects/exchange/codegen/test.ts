/* @hash 7f4682a38c988dedec87b805dda91de0 */
// tslint:disable
/* eslint-disable */
import { createWithContracts, TestOptions, WithContractsOptions } from '@neo-one/smart-contract-test';
import { Contracts } from './contracts';
import * as path from 'path';

export const withContracts: (
  test: (contracts: Contracts & TestOptions) => Promise<void>,
  options?: WithContractsOptions,
) => Promise<void> = createWithContracts([
  { name: 'Coin', filePath: path.resolve(__dirname, '../neo-one/contracts/Coin.ts') },
  { name: 'Exchange', filePath: path.resolve(__dirname, '../neo-one/contracts/Exchange.ts') },
  { name: 'Token', filePath: path.resolve(__dirname, '../neo-one/contracts/Token.ts') },
  { name: 'CoinICO', filePath: path.resolve(__dirname, '../neo-one/contracts/CoinICO.ts') },
  { name: 'ICO', filePath: path.resolve(__dirname, '../neo-one/contracts/ICO.ts') },
]);
