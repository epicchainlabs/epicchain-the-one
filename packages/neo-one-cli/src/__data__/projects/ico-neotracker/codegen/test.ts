/* @hash ba4c2d02de6f6fff287d33499e4c0951 */
// tslint:disable
/* eslint-disable */
import { createWithContracts, TestOptions, WithContractsOptions } from '@neo-one/smart-contract-test';
import { Contracts } from './contracts';
import * as path from 'path';

export const withContracts: (
  test: (contracts: Contracts & TestOptions) => Promise<void>,
  options?: WithContractsOptions,
) => Promise<void> = createWithContracts([
  { name: 'Escrow', filePath: path.resolve(__dirname, '../neo-one/contracts/Escrow.ts') },
  { name: 'Token', filePath: path.resolve(__dirname, '../neo-one/contracts/Token.ts') },
  { name: 'ICO', filePath: path.resolve(__dirname, '../neo-one/contracts/ICO.ts') },
]);
