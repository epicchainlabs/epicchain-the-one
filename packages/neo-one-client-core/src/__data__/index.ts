export * from './keys';
export * from './data';
export * from './factory';
export * from './HD';
export * from './verifyDataProvider';
