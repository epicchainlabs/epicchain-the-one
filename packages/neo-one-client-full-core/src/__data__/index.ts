export * from './keys';
export * from './data';
export * from './factory';
