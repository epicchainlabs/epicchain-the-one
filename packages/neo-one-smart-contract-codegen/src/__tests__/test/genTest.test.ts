import { contractsPaths } from '../../__data__/contractsPaths';
import { genTest } from '../../test';

describe('genTest', () => {
  test('Token', () => {
    expect(
      genTest({
        contractsPaths,
        contractsPath: '/foo/bar/one/generated/contracts.js',
        testPath: '/foo/bar/one/generated/test.js',
      }),
    ).toMatchSnapshot();
  });
});
