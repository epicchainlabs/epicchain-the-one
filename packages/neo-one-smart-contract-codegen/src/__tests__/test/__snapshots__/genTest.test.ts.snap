// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genTest Token 1`] = `
Object {
  "js": "
import { createWithContracts } from '@neo-one/smart-contract-test';
import * as path from 'path';

export const withContracts = createWithContracts([
  { name: 'Token', filePath: path.resolve(__dirname, '../contracts/Token.ts') }, { name: 'ICO', filePath: path.resolve(__dirname, '../contracts/ICO.ts') }
]);
",
  "ts": "
import { createWithContracts, TestOptions, WithContractsOptions } from '@neo-one/smart-contract-test';
import { Contracts } from './contracts';
import * as path from 'path';

export const withContracts: (
  test: (contracts: Contracts & TestOptions) => Promise<void>,
  options?: WithContractsOptions,
) => Promise<void> = createWithContracts([
  { name: 'Token', filePath: path.resolve(__dirname, '../contracts/Token.ts') }, { name: 'ICO', filePath: path.resolve(__dirname, '../contracts/ICO.ts') }
]);
",
}
`;
